### PR TITLE
[GNA] Draft of the plugin development documentation 

### PR DESCRIPTION
**Pull request by mryzhov**
### Details:
 - Added README.md and described the plugin design and structure

### Tickets:
 - *N/a*

### CODE REVIEW

File: src/plugins/intel_gna/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+* Outputs info
+* Pre/Post processing
+* Graph Compiler
+* Serializer
+* Memory Allocator
+* Request Factory
+* Device Helper
+
+### Network loading flow
+<img src="./docs/load_network_flow.png">
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gna/README.md:45]: how to support these pictures in future if something is changed in plugin architecture?
We use [mermaid](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams) to draw flows

File: src/plugins/intel_gna/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+* Denis Orlov <denis.orlov@intel.com>
+* Mikhail Ryzhov <mikhail.ryzhov@intel.com>
+* Evgeny Kotov <evgeny.kotov@intel.com>
+* Szymon Jakub Irzabek <szymon.jakub.irzabek@intel.com>
+* Nadezhda Ageeva <nadezhda.ageeva@intel.com>
+
+## Architecture
+
+### Plugin High Level Design
+<img src="./docs/hld.png" title="High level plugin architecture:">
```
Comment (marcinkusm) [inline src/plugins/intel_gna/README.md:25]: Just wonder if we could add arrows to present direction of interactions between components.

File: src/plugins/intel_gna/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    * [Add new transformation](#todo)
+ * [debug capabilities](./docs/debug_capabilities.md)
+ * [Code Style](./docs/code_style.md)
+
+## Key contacts
+* Denis Orlov <denis.orlov@intel.com>
+* Mikhail Ryzhov <mikhail.ryzhov@intel.com>
+* Evgeny Kotov <evgeny.kotov@intel.com>
+* Szymon Jakub Irzabek <szymon.jakub.irzabek@intel.com>
+* Nadezhda Ageeva <nadezhda.ageeva@intel.com>
```
Comment (sirzabek) [inline src/plugins/intel_gna/README.md:20]: Shouldn't we rather use a group here?

File: src/plugins/intel_gna/README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
```
Comment (marcinkusm) [inline src/plugins/intel_gna/README.md:21]: Just wonder if make sense to add following chapters e.g.:

## Glossary
| Name | Description|
|------|------------|
| GNA | Gaussian & Neural Accelerator |
| dnn | |
| CNN | |

## Delivered API
GNA plugin does not define API. It is dynamically loaded. Interface delivered by plugin is defined in.
[Openvion Dev API](https://github.com/marcinkusm/openvino/tree/master/src/inference/dev_api/cpp_interfaces/interface)

## Dependencies

### Build time dependencies
TODO
| Dependency | Purpose | comment |
|------------|---------|---------|
| ngraph | | |
| GNA library | | |

### Runtime Dependencies
TODO
| Dependency | Purpose | comment |
|------------|---------|---------|
| ngraph | | |
| GNA library | | |

File: src/plugins/intel_gna/README.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    src/                // Sources of the plugin
+    tests/              // Tests for the plugin
+        functional/     // Functional plugin tests
+        unit/           // Unit plugin tests
+    CMakeLists.txt      // Main CMake script
+    README.md           // Entry point for the developer documentation
+```
+
+
+## Features
```
Comment (marcinkusm) [inline src/plugins/intel_gna/README.md:64]: What do you think about adding use case diagrams?
image

File: src/plugins/intel_gna/README.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+ * [Import/Export](#todo)
+
+## See Also
```
Comment (marcinkusm) [inline src/plugins/intel_gna/README.md:67]: Some other suggested chapters e.g.:

# Open Issues
TODO

# Security Analysis
TODO

File: src/plugins/intel_gna/README.md (hunk 3 above)
Comment (mryzhov) [reply src/plugins/intel_gna/README.md:20]: Good idea, will change that

File: src/plugins/intel_gna/README.md (hunk 1 above)
Comment (mryzhov) [reply src/plugins/intel_gna/README.md:45]: mermaid a bit simple for such "pictures" (

**Comment by ilyachur** [comment]
@mryzhov Any updates for this PR?

**Comment by akladiev** [comment]
This PR will be closed in a week because of 2 weeks of no activity.

**Comment by akladiev** [comment]
This PR was closed because it has been stalled for 2 week with no activity.

**Comment by marcinkusm** [comment]
I see that in another PR [transformations doc](https://github.com/openvinotoolkit/openvino/pull/19232/files) the documentation is placed in docs subdirectory
image
IT would be good to put documentation in the same place.

**Comment by mryzhov** [comment]
> @mryzhov I see that in another PR [transformations doc](https://github.com/openvinotoolkit/openvino/pull/19232/files) the documentation is placed in docs subdirectory image IT would be good to put documentation in the same place.

let's merge PR with transformations doc and I will add to this PR then

**Pull request closed without merging**